src/InternalEvents.tsx

```diff
@@ -17,8 +17,9 @@ export default function InternalEvents() {
  * `useToggleHandler` handles the keyboard events for toggling kbar.
  */
 function useToggleHandler() {
-  const { query, options, visualState } = useKBar((state) => ({
+  const { query, options, visualState, hidden } = useKBar((state) => ({
     visualState: state.visualState,
+    hidden: state.visualState !== VisualState.showing,
```

Comment:
Probably just flip this to visible? As it's flipped here and flipped back in the guard